Gemfile

```diff
@@ -72,7 +72,7 @@ gem 'premailer-rails'
 gem 'rack-attack', '~> 6.6'
 gem 'rack-cors', '~> 1.1', require: 'rack/cors'
 gem 'rails-i18n', '~> 6.0'
-gem 'rails-settings-cached', '~> 0.6'
+gem 'rails-settings-cached', '~> 0.6', git: 'https://github.com/zunda/rails-settings-cached.git', branch: 'v0.6.6-aliases-true'
```

Comment:
I'm overall pretty worried about our dependency on rails-settings-cached considering that it's been an issue with almost every version upgrade. However, the newer version has major breaking changes and no clear way to migrate between versions. I think the data storage used by the most recent version (serialized hash or json data in a column on the accounts table) makes the most sense long-term, but at that point I'm not even sure what we're using a gem for lol. And certainly we'd still need something to handle the global site settings, which don't have an associated model they can live with.

Reply:
I studied the gem and didn't find a good way to upgrade either.  Then I looked in to what the gem does exactly.  It looks like it's storing app settings in the database, but using a YAML file for app defaults.  All values (from the database or YAML file) are also "cached" inside Redis too.

I was going to try changing the code that uses this to just use an in-memory cache because I couldn't understand the reason for putting things in Redis, but then I _think_ I stumbled on the use-case which is: when settings are updated the changes should be reflected across all web processes without restarting the app server.

I don't know if fetching these settings is a bottleneck, but it could be that just fetching them from Postgres every time would be fast enough.  Then we could just eliminate the gem.  I am not familiar with scaling Mastodon servers or what bottlenecks there are in the app, so I didn't want to just make that call.

Reply:
Agreed, and definitely out of scope for this PR, just something I think this raises that we should think about eventually.